pkg/startup/startup.go

```diff
@@ -421,19 +421,24 @@ func configureIPsAndSubnets(node *api.Node) (bool, error) {
 			log.Warnf("Autodetection of IPv6 address failed, keeping existing value: %s", node.Spec.BGP.IPv6Address)
 			validateIP(node.Spec.BGP.IPv6Address)
 		}
-	} else {
+	} else if ipv6Env != "none" {
 		if ipv6Env != "" {
 			node.Spec.BGP.IPv6Address = parseIPEnvironment("IP6", ipv6Env, 6)
 		}
 		validateIP(node.Spec.BGP.IPv6Address)
 	}
 
+	if ipv4Env == "none" && (ipv6Env == "" || ipv6Env == "none") {
+		log.Warn("No IP Addresses configured or set to autodetect")
+		terminate()
+	}
+
 	// Detect if we've seen the IP address change, and flag that we need to check for conflicting Nodes
-	if oldIpv4 == "" || node.Spec.BGP.IPv4Address != oldIpv4 {
+	if node.Spec.BGP.IPv4Address != oldIpv4 {
 		log.Info("Node IPv4 changed, will check for conflicts")
 		return true, nil
 	}
-	if (oldIpv6 == "" && node.Spec.BGP.IPv6Address != "") || (oldIpv6 != "" && node.Spec.BGP.IPv6Address != oldIpv6) {
```

Comment:
Weird... the old way just looks totally bonkers. Wonder how that happened.

Reply:
That's what I thought too. I look at it multiple times to figure out if it was handling some case that I wasn't thinking of.